app/common/data/interfaces/user.py

```diff
@@ -41,9 +41,12 @@ def get_users_with_permission(
     organisation_id: uuid.UUID | None | TNotProvided = NOT_PROVIDED,
     grant_id: uuid.UUID | None | TNotProvided = NOT_PROVIDED,
 ) -> Sequence[User]:
-    if organisation_id is None and grant_id is not None and grant_id is not NOT_PROVIDED:
-        raise ValueError("If specifying grant_id, must also specify organisation_id")
+    """Finds all users in the system with the given permission and matching organisation and/or grant.
 
+    NOTE: If you pass a grant but don't pass an organisation, this will only return users with the permission
+          *specifically* for that grant. It will not return users who have that permission in the grant through an
+          org-level permission (ie UserRole.grant_id IS NULL)
```

Comment:
As we're already set up for TNotProvided can we do 

```python
if grant_id is not NOT_PROVIDED and organisation_id is NOT_PROVIDED:
  organisation_id = grant.organistion_id
```

This feels a bit more intuitive to me, could remove this note and the nice thing about how you've set it up is you could still provide an explicit `organisation_id=None` to get only users with direct access, if you ever did need that

Reply:
Not totally sure I agree it's more intuitive, but I think one we can keep an eye on.